mockgcp/mocksql/sqlinstance.go

```diff
@@ -113,17 +113,26 @@ func (s *sqlInstancesService) Insert(ctx context.Context, req *pb.SqlInstancesIn
 	}
 
 	obj.GceZone = zone
-	obj.IpAddresses = []*pb.IpMapping{
-		{
-			IpAddress: "10.10.10.10",
-			Type:      pb.SqlIpAddressType_PRIMARY,
-		},
-	}
-	if isPostgres(obj) {
-		obj.IpAddresses = append(obj.IpAddresses, &pb.IpMapping{
-			IpAddress: "10.10.10.11",
-			Type:      pb.SqlIpAddressType_OUTGOING,
-		})
+	if obj.Settings.IpConfiguration.Ipv4Enabled.Value {
+		obj.IpAddresses = []*pb.IpMapping{
+			{
+				IpAddress: "10.10.10.10",
+				Type:      pb.SqlIpAddressType_PRIMARY,
+			},
+		}
+		if isPostgres(obj) {
+			obj.IpAddresses = append(obj.IpAddresses, &pb.IpMapping{
+				IpAddress: "10.10.10.11",
+				Type:      pb.SqlIpAddressType_OUTGOING,
+			})
+		}
+	} else {
+		obj.IpAddresses = []*pb.IpMapping{
+			{
+				IpAddress: "192.168.0.3",
```

Comment:
These IPs are currently hardcoded in the mockgcp implementation for SQLInstance. Should we consider extending the mock implementation to resolve ComputeAddress resource references (is this even possible?) and allocate IPs from their specified network ranges? Or, is this overkill? Not sure.